manager/controllers/app/fybrikmodule_controller.go

```diff
@@ -0,0 +1,140 @@
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If the object has a scheduled deletion time, update status and return
+	if !moduleContext.DeletionTimestamp.IsZero() {
+		// The object is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	observedStatus := moduleContext.Status.DeepCopy()
+	moduleVersion := moduleContext.GetGeneration()
+
+	// check if module has been validated before or if validated module is outdated
+	if string(moduleContext.Status.ValidModule) == "" || observedStatus.ValidatedGeneration != moduleVersion {
+		// do validation on moduleContext
+		err := ValidateFybrikModule(moduleContext, "/tmp/taxonomy/fybrik_module.json")
+		log.V(0).Info("Reconciler validating Fybrik module")
+		moduleContext.Status.ValidatedGeneration = moduleVersion
+		// if validation fails
+		if err != nil {
+			// set error message
+			log.V(0).Info("Fybrik module validation failed " + err.Error())
+			moduleContext.Status.ErrorMessage = err.Error()
+			moduleContext.Status.ValidModule = v1.ConditionFalse
+		} else {
+			moduleContext.Status.ValidModule = v1.ConditionTrue
+			moduleContext.Status.ErrorMessage = ""
+		}
+	}
+
+	// Update CRD status in case of change (other than deletion, which was handled separately)
+	if !equality.Semantic.DeepEqual(&moduleContext.Status, observedStatus) && moduleContext.DeletionTimestamp.IsZero() {
+		log.V(0).Info("Reconcile: Updating status for desired generation " + fmt.Sprint(moduleContext.GetGeneration()))
+		if err := r.Client.Status().Update(ctx, moduleContext); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// NewFybrikModuleReconciler creates a new reconciler for FybrikModules
+func NewFybrikModuleReconciler(mgr ctrl.Manager, name string) *FybrikModuleReconciler {
+	return &FybrikModuleReconciler{
+		Client: mgr.GetClient(),
+		Name:   name,
+		Log:    ctrl.Log.WithName("controllers").WithName(name),
+		Scheme: mgr.GetScheme(),
+	}
+}
+
+func ValidateFybrikModule(module *api.FybrikModule, taxonomyFile string) error {
+	var allErrs []*field.Error
+
+	// Convert Fybrik module Go struct to JSON
+	applicationJSON, err := json.Marshal(module)
+	if err != nil {
+		return err
+	}
+
+	// Validate Fybrik module against taxonomy
+	allErrs = validate.TaxonomyCheck(applicationJSON, taxonomyFile, "Fybrik module")
+
+	// Return any error
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return apierrors.NewInvalid(
+		schema.GroupKind{Group: "app.fybrik.io", Kind: "FybrikModule"},
+		module.Name, allErrs)
+}
+
+// SetupWithManager registers Module controller
+func (r *FybrikModuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// 'UpdateFunc' and 'CreateFunc' used to judge if the event came from within the blueprint's namespace.
+	// If that is true, the event will be processed by the reconciler.
+	// If it's not then it is a rogue event created by someone outside of the control plane.
+
+	moduleNamespace := utils.GetSystemNamespace()
+
+	p := predicate.Funcs{
```

Comment:
@shlomitk1 Am I correct that we don't need this, because you addressed this through RBAC configurations?

Reply:
Right, we don't need this.

Reply:
Removed `p`